topdown/uuid.go

```diff
@@ -31,6 +32,32 @@ func builtinUUIDRFC4122(bctx BuiltinContext, operands []*ast.Term, iter func(*as
 	return iter(result)
 }
 
+func builtinUUIDParse(_ BuiltinContext, operands []*ast.Term, iter func(term *ast.Term) error) error {
+	str, err := builtins.StringOperand(operands[0].Value, 1)
+	if err != nil {
+		return err
+	}
+
+	parsed, err := uuid.Parse(string(str))
+	if err != nil {
+		return nil
+	}
+
+	result := ast.NewObject()
+	for prop, val := range parsed {
+		switch val := val.(type) {
+		case string:
+			result.Insert(ast.StringTerm(prop), ast.StringTerm(val))
+		case int:
+			result.Insert(ast.StringTerm(prop), ast.IntNumberTerm(val))
+		case int64:
+			result.Insert(ast.StringTerm(prop), ast.NumberTerm(int64ToJSONNumber(val)))
+		}
+	}
+	return iter(ast.NewTerm(result))
```

Comment:
I think you could also use `ast.InterfaceToValue` to construct `result` from `parsed`; but this is OK, too.

Reply:
Did not see that function😮 That is waaaay better, and makes it way easier. I like it. Changed it now so it is like two lines now instead